hazelcast/src/hazelcast/client/serialization/TypeIDS.cpp

```diff
@@ -66,6 +66,46 @@ namespace hazelcast {
             int32_t getHazelcastTypeId(const std::string *object) {
                 return pimpl::SerializationConstants::CONSTANT_TYPE_STRING;
             }
+
+            int32_t getHazelcastTypeId(const std::vector<char> *object) {
+                return pimpl::SerializationConstants::CONSTANT_TYPE_CHAR_ARRAY;
+            }
+
+            int32_t getHazelcastTypeId(const std::vector<bool> *object) {
+                return pimpl::SerializationConstants::CONSTANT_TYPE_BOOLEAN_ARRAY;
+            }
+
+            int32_t getHazelcastTypeId(const std::vector<byte> *object) {
+                return pimpl::SerializationConstants::CONSTANT_TYPE_BYTE_ARRAY;
+            }
+
+            int32_t getHazelcastTypeId(const std::vector<int16_t> *object) {
+                return pimpl::SerializationConstants::CONSTANT_TYPE_SHORT_ARRAY;
+            }
+
+            int32_t getHazelcastTypeId(const std::vector<int32_t> *object) {
+                return pimpl::SerializationConstants::CONSTANT_TYPE_INTEGER_ARRAY;
+            }
+
+            int32_t getHazelcastTypeId(const std::vector<int64_t> *object) {
+                return pimpl::SerializationConstants::CONSTANT_TYPE_LONG_ARRAY;
+            }
+
+            int32_t getHazelcastTypeId(const std::vector<float> *object) {
+                return pimpl::SerializationConstants::CONSTANT_TYPE_FLOAT_ARRAY;
+            }
+
+            int32_t getHazelcastTypeId(const std::vector<double> *object) {
+                return pimpl::SerializationConstants::CONSTANT_TYPE_DOUBLE_ARRAY;
+            }
+
+            int32_t getHazelcastTypeId(const std::vector<std::string *> *object) {
+                return pimpl::SerializationConstants::CONSTANT_TYPE_STRING_ARRAY;
```

Comment:
std::vector<std::string *> and  std::vector<std::string> serializers should have different id's. Otherwise you will get illegal cast here
https://github.com/hazelcast/hazelcast-cpp-client/pull/336/files#diff-8c0e1e3a5025fea24f26381129c53b1eR248